### ISSUE TITLE
Waiting for root device.

### ISSUE THREAD

**Issue by spauldhaliwal** [syscl/XPS9350-macOS#179]
I get "waiting for root device" error when I use the clover folder from this repository when trying to run the installer. Installing High Sierra on the USB using unibeast without your clover folder I am able to boot into the installer. Will I run into issues down the road having not used your clover folder? Thanks.

**Comment by spauldhaliwal** [comment]
I've discovered that for some reason I can only boot into the installer using your Clover folder if I copy all of the kext files from the '10.13' into a new folder titled 'Other'.

**Comment by ggctseng** [comment]
@spauldhaliwal I'm not sure if this is the issue, but I think the installer is not "versioned" like regular OS X so it will load kexts from the "Other" folder.  I'm pretty sure you can just leave the "required" kexts like FakeSMC and VoodoPS2 in the Other folder (and everything else in the 10.13 folder) if it bothers you?  Please test and confirm if you get a chance.

BTW, to answer your earlier question, you don't have to use the CLOVER folder on the repository for installing OS X (I actually wasn't able to do so and used a "generic" CLOVER folder--if you want to see more details, see the other thread).

**Comment by spauldhaliwal** [comment]
I eventually got it up and running by putting all the kexts in Other for install. However, I then encountered a bootloop after install was finished. That was fixed by changing SMBIOS to MacBookAir6,2.

Everything seems to be working fine now except for the trackpad. Around 20% of taps don't register and I can't seem to change any of the gesture settings. I'd like to have three finger swipe left/right to switch between full-screen apps and desktops however changing that setting doesn't have any effect. In fact turning off all the gestures in the system settings has no effect and all work as default :S. 

Also two finger scrolling works fine in finder however in safari (and Opera which is my browser of choice) it is very choppy and slow. Is this normal or must there be something wrong with my setup? Thanks!

**Comment by ggctseng** [comment]
@spauldhaliwal What version of the Dell bios are you on (if 1.6.1, what date was the bios--they pulled an earlier version and released a new one on 2/5/18 or something)? Are you using the Voodoo driver?

I just tried my trackpad in Safari, and it works.  3-finger gestures seem to work too (although the trackpad is really small so I don't use 3-finger gestures--and I have small hands!).

![screen shot 2018-02-20 at 11 18 15 am](https://user-images.githubusercontent.com/9076368/36444271-ddbf83fa-162f-11e8-828f-7ed012e98b48.png)

I reported before that I had issues with 13,3 identifier in the SMBIOS and MacBookPro13,1 works for me (according to [everymac.com](https://everymac.com/systems/apple/macbook_pro/specs/macbook-pro-core-i7-2.4-13-late-2016-retina-display-no-touch-bar-specs.html), MacBookPro13,1 is an almost perfect match for the XPS 9350 except for the processor, which is slightly faster on the MacBook).  Please try using that identifier and report back.  If you're feeling particularly saucy, you can try [MacBookAir7,2](https://everymac.com/systems/apple/macbook-air/specs/macbook-air-core-i7-2.2-13-early-2015-specs.html) as the identifier, which has the earlier (Broadwell) version of the processor but is a frequency match to the i7-6560 in the XPS (I didn't check the i5 version).  Of course, neither MacBookAir6,2 or MacBookAir7,2 has TB3 (they both have TB2). **_NOTE: I haven't tried using MacBookAir7,2 as the identifier, so..._**

BTW, during testing you should log out of iCloud/Messages/FaceTime and you'll want(need?) to regenerate serials, etc. once you switch to a new identifier.  I think you can just re-run the script once you change the config.plist with the different identifier(although you should definitely back up your current one).

**Comment by spauldhaliwal** [comment]
@ggctseng I'm on 1.5.1 of the bios. I'm using the voodoo kext that was installed by Deploy.sh.

I'll try that smbios identier tonight when I'm home. Would that have an effect on something like trackpad performance? Would you be able to check if changing gesture settings has any effect for you? Such as disabling them? I unticked all the gestures in 'more gestures' as a test and it had no effect. All gestures continued as default.

I don't use iCloud/iMessage/FaceTime so I'm not too worries about that haha.

Thanks for the suggestions! I'll try them tonight and report back.

**Comment by ggctseng** [comment]
@spauldhaliwal I don't think the identifier would have an effect. However, I'm way beyond being surprised by whether something is affected or not by a seemingly unrelated change. 

With regard to track pad performance, I'm sorry, but I think I mislead you. I actually tried various multi-finger actions and:

"Point & Click":  Look up is unaffected by the Trackpad setting.  The Secondary click and Tap to click changes when the options are changed.
"Scroll & Zoom": None of these options seem to work.
"More Gestures": Mission Control works, but only the swiping up part and swiping down does nothing like how it demos in the pref. panel (I actually have to swipe up again to "undo" Mission Control. I got Show Desktop to work once, lol. Nothing else works. Also changing options (or unchecking boxes) does nothing.  For example, even if I uncheck Mission Control I can still swipe up with 3-fingers and Mission Control will activate.

Let me know how you get along with changing the identifier.  I don't have time to play with it anymore.  I ran the new deploy.sh, but I think the issue was there before.  TBH, I don't use the Trackpad features at all so I wouldn't notice (I generally only use the Point & Click page, LOL.

How could you not use iMessage?  I guess you don't have an iPhone, hahaha.

**Comment by spauldhaliwal** [comment]
Yep I tried the MacBookPro13,1 Identifier and it worked. I had to delete a couple of lines from config.plist for it to take effect but it eventually took. I also noticed that for the first stage of loading the OS the apple logo is huge, and then only becomes normal size when the screen refreshes (loads display drivers I suppose?)

It's strange that some settings are effected by changes and others aren't. I'd really like to be able to use three finger swipes to switch between spaces, as I don't use back and forward pages functions ever and also the trackpad is too small for four finger gestures I find haha.

And no, no iPhone here haha.

**Comment by spauldhaliwal** [comment]
@ggctseng I'm also just wondering in general if my trackpad is behaving similar to other's with this Voodoo kext. Would you say it compares in performance to on Windows? To me it seems like 20% of the time single taps to click don't register and gesture recognition is way less sensitive than on Windows. Is this normal behaviour? And I wonder if it can be tuned somehow. I think I'll open up a new issue since this doesn't really relate to the orignal post anymore haha.

**Comment by spauldhaliwal** [comment]
So I've discovered that these gesture's seem to only send key combinations. So simply editing keyboard shortcuts and then performing the gestures you want configured seems to do the trick!

**Comment by ggctseng** [comment]
Do you have a link that lists what text combinations the gestures are sending? I use BetterTouchTool on my real Mac, but it doesn't recognize the trackpad on the XPS as multi-touch.

**Comment by ggctseng** [comment]
@spauldhaliwal Oops, forgot to direct my previous comment towards you, but it seems like we're the only active ones on this thread anyway.

YES, the whole smbios identifier is a crap shoot.
YES, the XPS trackpad is too small, and really after using the MacBook trackpad, size notwithstanding, the XPS trackpad just can't compare.  I don't remember what it was like in Windows as I barely had Windows 10 on it. TBH I kind of regret buying it because it's actually too small and not "hefty" enough.

**Comment by spauldhaliwal** [comment]
Nope don't have a list but if you go to record a keyboard shortcut in better touch tool and perform the gesture it will record they key combinations. For me now that I have the gestures setup the trackpad is working pretty well. The only issue I still have is often 'Single tap to click' doesn't register, which is pretty annoying. Does that happen to you ever?

**Comment by ggctseng** [comment]
@spauldhaliwal Sorry, haven't had time to test as I've been traveling. Will try to test and revert this weekend.

**Comment by hebert1410** [comment]
I have the same situation happening to me recently when I was trying to install High Sierra in my XPS 9350. I followed all the instructions step by step provided by syscl but for some reason I keep getting the "still waiting for root device" crashing half way the booting. Then, I saw this thread and   moved the kext files from the 10.13 folder to Others and I'm still encountering the same problem. I don't think it is the USB drive or the method of creating the installation drive because I tried it in multiple drives and methods (Unibeast and Terminal commands) and so far no changes. I also made sure I had the last version of Clover. My current set up is Intel i5 Skylake, 8gb ram, 512 SSD, 1830 Broadcom, touchscreen QHD+ with 1.5.1 BIOS installed.
